app/src/main/java/org/oppia/app/parser/StringToRatioParser.kt

```diff
@@ -0,0 +1,85 @@
+package org.oppia.app.parser
+
+import android.content.Context
+import androidx.annotation.StringRes
+import org.oppia.app.R
+import org.oppia.app.model.RatioExpression
+
+/** This class contains method that helps to parse string to ratio. */
+class StringToRatioParser {
+  private val invalidCharsRegex =
+    """^[\d\s/:]+$""".toRegex()
+  private val invalidRatioRegex =
+    """^(\s)*(\d+((\s)*:(\s)*\d+)+)(\s)*$""".toRegex()
+
+  /**
+   * Returns a [RatioParsingError] for the specified text input if it's an invalid ratio, or
+   * [RatioParsingError.VALID] if no issues are found. Note that a valid ratio returned by this method is guaranteed
+   * to be parsed correctly by [parseRatioFromString].
```

Comment:
``parseRatioFromString`` is no longer the correct method name, please fix.

Also, doesn't this show as an error for you in Android Studio? I'm curious how this was missed.

Reply:
updated